internal/users/users.go

```diff
@@ -16,3 +18,23 @@ type Group struct {
 	Name string
 	GID  *int
 }
+
+func UsersFromBP(userCustomizations []blueprint.UserCustomization) []User {
+	users := make([]User, len(userCustomizations))
+	for idx := range userCustomizations {
+		// currently, they have the same structure, so we convert directly
+		// this will fail to compile as soon as one of the two changes
+		users[idx] = User(userCustomizations[idx])
+	}
+	return users
+}
+
+func GroupsFromBP(groupCustomizations []blueprint.GroupCustomization) []Group {
+	groups := make([]Group, len(groupCustomizations))
+	for idx := range groupCustomizations {
+		// currently, they have the same structure, so we convert directly
+		// this will fail to compile as soon as one of the two changes
+		groups[idx] = Group(groupCustomizations[idx])
+	}
+	return groups
+}
```

Comment:
Did you consider making these a method of the respective `blueprint.XXXCustomization` types?

Reply:
This package is an analogue to the `disk` and `ostree` packages.  In `disk` we do similar things where we take the `FilesystemCustomization` from the blueprint (and some other options) to generate a `PartitionTable`.
It's also similar to how we do `GenXStage()` functions in osbuild, taking some representation of the options and generating the stage.

It's not a rule, but I think we're mostly consistent with these kinds of transformation/generator functions live in the package of the types they generate.

Reply:
Fair enough...

I mean, both approaches are valid. I incline to the OOP approach and implementing functions that work only with a particular data as a methods of that type.